app/src/main/java/de/westnordost/streetcomplete/quests/max_weight/AddMaxWeightForm.kt

```diff
@@ -0,0 +1,175 @@
+        setMaxWeightSignLayout(R.layout.quest_maxweight, unit)
+
+        return view
+    }
+
+    private fun setMaxWeightSignLayout(resourceId: Int, unit: Measurement) {
+        val contentView = setContentView(resourceId)
+
+        tonInput = contentView.findViewById(R.id.tonInput)
+        poundInput = contentView.findViewById(R.id.poundInput)
+
+        val onTextChangedListener = TextChangedWatcher { checkIsFormComplete() }
+        tonInput?.addTextChangedListener(onTextChangedListener)
+        poundInput?.addTextChangedListener(onTextChangedListener)
+
+        tonInputSign = contentView.findViewById(R.id.tonInputSign)
+        poundInputSign = contentView.findViewById(R.id.poundInputSign)
+
+        weightUnitSelect = contentView.findViewById(R.id.weightUnitSelect)
+        val measurementUnits = countryInfo.measurementSystem
+        weightUnitSelect?.visibility = if (measurementUnits.size == 1) View.GONE else View.VISIBLE
+        weightUnitSelect?.adapter = ArrayAdapter(context!!, R.layout.spinner_item_centered, getSpinnerItems(measurementUnits))
+        weightUnitSelect?.setSelection(measurementUnits.size - 1) // prefer metric
+        weightUnitSelect?.onItemSelectedListener = object : AdapterView.OnItemSelectedListener {
+            override fun onItemSelected(parentView: AdapterView<*>, selectedItemView: View?, position: Int, id: Long) {
+                val heightUnit = if (weightUnitSelect?.selectedItem == "t") METRIC else IMPERIAL
+                switchLayout(heightUnit)
+            }
+
+            override fun onNothingSelected(parentView: AdapterView<*>) {}
+        }
+
+        /* Workaround for an Android bug that it assumes the decimal separator to always be the "."
+           for EditTexts with inputType "numberDecimal", independent of Locale. See
+           https://issuetracker.google.com/issues/36907764 .
+
+           Affected Android versions are all versions till (exclusive) Android Oreo. */
+
+        /* actually, let's not care about which separator the user uses, he might be confused
+           whether he should use the one as displayed on the sign or in his phone's locale */
+        //char separator = DecimalFormatSymbols.getInstance(getCountryInfo().getLocale()).getDecimalSeparator();
+        tonInput?.keyListener = DigitsKeyListener.getInstance("0123456789,.")
+
+        switchLayout(unit)
+    }
+
+    private fun switchLayout(unit: Measurement) {
+        val isMetric = unit == METRIC
+        val isImperial = unit == IMPERIAL
+
+        tonInputSign?.visibility = if (isMetric) View.VISIBLE else View.GONE
+        poundInputSign?.visibility = if (isImperial) View.VISIBLE else View.GONE
+
+        if (isMetric) tonInput?.requestFocus()
+        if (isImperial) poundInput?.requestFocus()
+    }
+
+    private fun getSpinnerItems(units: List<String>) = units.mapNotNull {
+        when(it) {
+            "metric" -> "t"
+            "imperial" -> "lbs"
+            else -> null
+        }
```

Comment:
Confused here: What about the US sign that should have the short tons and pounds in the spinner?

Reply:
US sign have short tons described as "t" so it is also displayed this way and later converted to `st` for US locations.

Reply:
Then it should not be called "metric" there, otherwise the code is confusing to read.

Reply:
> handle US units - currently it is correctly set to imperial in https://github.com/westnordost/StreetComplete/blob/master/res/country_metadata/measurementSystem.yml#L38 but to enable both short ton and pound units it would need to be [imperial, metric] (incorrect for height quest) or have separate metadata file or have special handling somewhere

Maybe https://github.com/westnordost/StreetComplete/blob/master/res/country_metadata/measurementSystem.yml#L38 should have 

> US: [imperial, short_ton]

or

> US: [imperial, US]

? Or is special hardcoding in quest (and overriding country_metadata for US) better?

Either way US is a special case, not sure what is better - explicit declaring it in the metada file or not, either way special case for US needs to be implemented in some places.

Reply:
You mean, only in the US, there is pounds AND short tons, and in other countries that use imperial measure, there is just short tons? If yes, what makes you arrive at that conclusion? To me, it would be most logical to both allow pounds (lbs) and short tons as input for countries that use the imperial measure.

Reply:
It probably would make sense, though for now US is only place that has units that is not ton. Other places using imperial units appear to be using tons for max weight (or are so small that have no bridges).

Reply:
What is the source of this information?

Reply:
> What is the source of this information?

For countries using imperial units I looked through official sites/Wikipedia lists of traffic signs/asked local mappers via mailing lists/slack.

Reply:
Also, this is probably a problem. Doesn't GB use metric tonnes for measurement now?

https://github.com/westnordost/StreetComplete/blob/master/res/country_metadata/measurementSystem.yml#L19

Probably weight and lengths need to be defined separately.

Reply:
> Doesn't GB use metric tonnes for measurement now?

Yes, for example GB switched (except some crown dependencies that are tiny islands where data is not triggering this quest, and in case of building new one it can be solved with a note).

Reply:
Okay, so the measurement system for lengths and for weights need to be defined separately. Then one option would be to call these files `weightUnits` and `lengthUnits` instead, and use values like `st`, `lbs`, `meter` and so on instead of the more abstract `imperial` and `metric`

Reply:
Should be fixed now or at least improved.